packages/web3torrent/src/clients/payment-channel-client.ts

```diff
@@ -258,15 +259,13 @@ export class PaymentChannelClient {
         state.payer === this.mySigningAddress &&
         state.turnNum.mod(2).eq(Index.Payer);
 
-      const currentState = this.channelCache[channelId];
-      if (readyToPay(currentState)) resolve(currentState);
-
-      const unsubscribeListener = this.channelClient.onChannelUpdated(cu => {
-        if (readyToPay(convertToChannelState(cu))) {
-          unsubscribeListener();
-          resolve(convertToChannelState(cu));
-        }
-      });
+      this.channelClient.channelState
+        .pipe(
+          map((result: ChannelResult) => convertToChannelState(result)),
+          takeWhile((latestChannelState: ChannelState) => !readyToPay(latestChannelState), true),
+          filter(readyToPay)
```

Comment:
[takeLast](https://rxjs-dev.firebaseapp.com/api/operators/takeLast) might be more clear

Reply:
take a look at https://github.com/statechannels/monorepo/pull/1521/commits/b31af613ca8f8d85e206bea2c401cc792a19a5c1

Reply:
Thanks!